app/components/custom_fields/details_component.rb

```diff
@@ -50,5 +51,16 @@ def form_url
     def form_method
       model.new_record? ? :post : :put
     end
+
+    def enterprise_addon
+      @enterprise_addon ||= case custom_field.field_format
+                            when "hierarchy"
+                              { key: :custom_field_hierarchies, image: "enterprise/hierarchies.png" }
+                            when "weighted_item_list"
+                              { key: :weighted_item_lists, image: "enterprise/weighted_item_lists.png" }
+                            else
+                              {}
```

Comment:
:red_circle: This will cause errors in the attempt of rendering the enterprise banner (because we'd try to render a banner with a `nil` key and a missing image.

I'd like `{}` as a return value, if it would cause a useful default behaviour. Since it causes broken rendering though, we should raise an error here directly.

Proposal:

```suggestion
                              raise "Custom fields of format #{custom_field.field_format} are not supported by #{self.class.name}"
```

Reply:
I get ya, will raise instead of return empty!